pkg/controller/daemon/daemon_controller.go

```diff
@@ -1040,7 +1040,7 @@ func storeDaemonSetStatus(dsClient unversionedapps.DaemonSetInterface, ds *apps.
 	toUpdate := ds.DeepCopy()
 
 	var updateErr, getErr error
-	for i := 0; i < StatusUpdateRetries; i++ {
+	for i := 0; ; i++ {
 		if updateObservedGen {
 			toUpdate.Status.ObservedGeneration = ds.Generation
 		}
```

Comment:
As if updateObserveGen has been called in storeDaemonSetStatus function :
1041 
   1042         var updateErr, getErr error
   1043         for i := 0; i < StatusUpdateRetries; i++ {
   1044                 if updateObservedGen {
   1045                         toUpdate.Status.ObservedGeneration = ds.Generation
   1046                 }

Reply:
Sorry, could you explain what you mean?